### ISSUE TITLE
prefer-nullish-coalescing reports Linux-only false positives for boolean OR expressions

### ISSUE THREAD

**Issue by bk-tho** [oxc-project/tsgolint#1025]
### Summary

`oxlint` with `oxlint-tsgolint` reports `typescript/prefer-nullish-coalescing` errors in Linux CI, but the same repository, lockfile, and lint command pass locally on macOS.

Some reported diagnostics appear to be false positives where replacing `||` with `??` would change semantics, especially for boolean expressions.

### Versions

- `oxlint`: `1.71.0`
- `oxlint-tsgolint`: `0.23.0`
- TypeScript native package: `typescript@7.0.1-rc`
- Local platform: macOS arm64, `@oxlint-tsgolint/darwin-arm64@0.23.0`
- CI platform: Linux x64 Docker image, expected `@oxlint-tsgolint/linux-x64@0.23.0`

### Config

```jsonc
{
  "options": {
    "typeAware": true
  },
  "plugins": ["typescript"],
  "rules": {
    "typescript/prefer-nullish-coalescing": "error"
  }
}
```

### Command

```sh
bunx oxlint -c .oxlintrc.json
```

### CI Diagnostics

The Linux CI run reports errors like these:

```tsx
const showColumns = isRelevantPosition || hasValues;
```

Diagnostic:

```text
typescript(prefer-nullish-coalescing): Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator.
```

This suggested replacement appears incorrect because both operands are booleans. Replacing with `??` would change behavior when `isRelevantPosition` is `false` and `hasValues` is `true`.

### Expected Behavior

The rule should not report `||` expressions where `??` would change semantics, such as:

- boolean OR expressions
- truthiness checks for strings where empty string behavior matters
- conditional rendering guards that intentionally use JavaScript truthiness

The rule should also behave consistently across platform-specific `oxlint-tsgolint` binaries.

### Actual Behavior

- macOS arm64 local lint: `Found 0 warnings and 0 errors.`
- Linux x64 CI lint: reports `typescript/prefer-nullish-coalescing` errors for the examples above.

**Comment by camc314** [comment]
Sorry for the delay! this one got lost.

do you have any additional context to help reproduce this issue?

Can you run with `--type-check` to see if there's TS errors that differs on mac vs linux as this may be the cause of the issue.

Thanks!

**Comment by camc314** [comment]
Closing as not planned - can't repro.
